File: app/src/main/java/com/canchapp_kotlin/data/network/ApiModels.kt
package com.canchapp_kotlin.data.network

import com.google.gson.annotations.SerializedName

data class LoginRequest(
    val identifier: String,
    val password: String
)

data class LoginResponse(
    val success: Boolean,
    val message: String?,
    val data: LoginData?,
    val error: ApiError?
)

data class LoginData(
    val access: String,
    val refresh: String,
    val user: UserDto
)

data class RegisterRequest(
    val email: String,
    @SerializedName("f_name") val firstName: String,
    @SerializedName("l_name") val lastName: String,
    val password: String,
    @SerializedName("role_name") val roleName: String,
    val username: String
)

data class RegisterResponse(
    val success: Boolean,
    val message: String?,
    val data: UserDto?,
    val error: ApiError?
)

data class UserDto(
    @SerializedName("user_id")    val userId: String,
    val username: String,
    val email: String,
    @SerializedName("f_name")     val firstName: String,
    @SerializedName("l_name")     val lastName: String,
    @SerializedName("role_name")  val roleName: String,
    val status: String,
    @SerializedName("avatar_url") val avatarUrl: String?,
    @SerializedName("is_guest")   val isGuest: Boolean
)

data class ApiError(
    val code: String?,
    val message: String?,
    val details: Map<String, Any>?
)

data class ComplexDto(
    @SerializedName("complex_id")   val complexId: String,
    @SerializedName("owner_id")     val ownerId: String,
    val name: String,
    val address: String,
    val city: String,
    val latitude: Double,
    val longitude: Double,
    val status: String,
    @SerializedName("fields_count") val fieldsCount: Int,
    @SerializedName("min_price")    val minPrice: Double,
    @SerializedName("max_price")    val maxPrice: Double,
    @SerializedName("distance_km")  val distanceKm: Double?
)

data class ComplexListData(
    val items: List<ComplexDto>,
    val total: Int,
    val page: Int,
    @SerializedName("page_size")    val pageSize: Int,
    @SerializedName("total_pages")  val totalPages: Int
)

data class ComplexListResponse(
    val success: Boolean,
    val message: String?,
    val data: ComplexListData?,
    val error: ApiError?
)

File: ios/CanchApp/Models/ApiModels.swift
import Foundation

// MARK: - Login

struct LoginRequest: Encodable {
    let identifier: String
    let password: String
}

struct LoginResponse: Decodable {
    let success: Bool
    let message: String?
    let data: LoginData?
    let error: ApiError?
}

struct LoginData: Decodable {
    let access: String
    let refresh: String
    let user: UserDto
}

// MARK: - Register

struct RegisterRequest: Encodable {
    let email: String
    let firstName: String
    let lastName: String
    let password: String
    let roleName: String
    let username: String

    enum CodingKeys: String, CodingKey {
        case email, password, username
        case firstName = "f_name"
        case lastName  = "l_name"
        case roleName  = "role_name"
    }
}

struct RegisterResponse: Decodable {
    let success: Bool
    let message: String?
    let data: UserDto?
    let error: ApiError?
}

// MARK: - User

struct UserDto: Codable {
    let userId: String
    let username: String
    let email: String
    let firstName: String
    let lastName: String
    let roleName: String
    let status: String
    let avatarUrl: String?
    let isGuest: Bool

    enum CodingKeys: String, CodingKey {
        case username, email, status
        case userId    = "user_id"
        case firstName = "f_name"
        case lastName  = "l_name"
        case roleName  = "role_name"
        case avatarUrl = "avatar_url"
        case isGuest   = "is_guest"
    }
}

// MARK: - Error

struct ApiError: Decodable {
    let code: String?
    let message: String?
}

// MARK: - Complex

struct ComplexDto: Codable, Identifiable {
    var id: String { complexId }
    let complexId: String
    let ownerId: String
    let name: String
    let address: String
    let city: String
    let latitude: Double
    let longitude: Double
    let status: String
    let fieldsCount: Int
    let minPrice: Double
    let maxPrice: Double
    let distanceKm: Double?

    enum CodingKeys: String, CodingKey {
        case name, address, city, latitude, longitude, status
        case complexId   = "complex_id"
        case ownerId     = "owner_id"
        case fieldsCount = "fields_count"
        case minPrice    = "min_price"
        case maxPrice    = "max_price"
        case distanceKm  = "distance_km"
    }
}

struct ComplexListData: Decodable {
    let items: [ComplexDto]
    let total: Int
    let page: Int
    let pageSize: Int
    let totalPages: Int

    enum CodingKeys: String, CodingKey {
        case items, total, page
        case pageSize   = "page_size"
        case totalPages = "total_pages"
    }
}

struct ComplexListResponse: Decodable {
    let success: Bool
    let message: String?
    let data: ComplexListData?
    let error: ApiError?
}
